### DISCUSSION TITLE
prevent hashing columns with null values

### DISCUSSION THREAD

**Discussion by aryala7** [spatie/laravel-ciphersweet#31] [Ideas]
is it possible to prevent hashing null values? for example if one of the user's cellphones is null the value remain null as well.
 i would like to create a PR and add this as an optional param in ciphersweet.php config if its valid

**Comment by mikerockett** [comment]
I'd like to see this as well. For now, I'm lucky in that I don't need to call `whereNull` on the column in question. As a last resort, I can create an additional column that indicates whether or not the column in question is populated, but that isn't really ideal.

**Comment by yormy** [comment]
When I do $encryptedRow->addOptionalTextField('field')
I expect that the value is only encrypted if not NULL, however it does encrypt the NULL value also

**Comment by yormy** [comment]
I created a Pull Request that fixes this null encryption

**Comment by yormy** [comment]
Has been merged
